### PR TITLE
Addition of Gender Method Tests

### PR DESCRIPTION
**Pull request by WVinson7**
testGetGender()
testSetGender()
testSetGenderBad()
testSetGenderBadMaleToFemale()

### CODE REVIEW

File: src/test/java/edu/kcc/animal/AnimalTest.java [hunk 1]
```diff
@@ -107,26 +110,53 @@ public void testSetSpeciesBadCatToDog() {
...
+        assertEquals(gender, animal.getGender());
     }
     
     // TODO - Attempt to set non male or female
     @Test
     public void testSetGenderBad() {
-        fail("The test case is a prototype.");
+        setUp();
+        String gender = "vegetable";
+        String original = GOOD_GENDER;
```
Comment (mlhaus) [inline src/test/java/edu/kcc/animal/AnimalTest.java:134]: The original gender is actually unknown. You would have to set the gender to GOOD_GENDER first to have this work.

File: src/test/java/edu/kcc/animal/AnimalTest.java [hunk 2]
```diff
@@ -107,26 +110,53 @@ public void testSetSpeciesBadCatToDog() {
...
      // TODO - Attempt to set a male to female
     @Test
     public void testSetGenderBadMaleToFemale() {
-        fail("The test case is a prototype.");
+        setUp();
+        String originalEntry = "male";
+        animal.setGender(originalEntry);
+        try{
+            animal.setGender(GOOD_GENDER);
+            if(animal.getGender() == originalEntry){
```
Comment (mlhaus) [inline src/test/java/edu/kcc/animal/AnimalTest.java:151]: Substitute the if statement with a fail statement -- if the program allows us to set the gender from male to female we need the test to fail.

**Review by mlhaus — CHANGES_REQUESTED** [review]
See comments for requested changes.

File: src/test/java/edu/kcc/animal/AnimalTest.java (hunk 1 above)
Comment (WVinson7) [reply src/test/java/edu/kcc/animal/AnimalTest.java:136]: @Test
    public void testSetGenderBad() {
        setUp();
        String gender = "vegetable";
        try{
            animal.setGender(gender);
            fail("Gender not valid.");
        }
        catch (IllegalArgumentException iae){
            assertEquals(gender, animal.getGender());
        }
    }

Like this?

File: src/test/java/edu/kcc/animal/AnimalTest.java (hunk 2 above)
Comment (WVinson7) [reply src/test/java/edu/kcc/animal/AnimalTest.java:153]: @Test
    public void testSetGenderBad() {
        setUp();
        String gender = "vegetable";
        try{
            animal.setGender(gender);
            fail("Gender not valid.");
        }
        catch (IllegalArgumentException iae){
            assertEquals(gender, animal.getGender());
        }
    }

The validator wasn't written yet. Now it is! 👍

**Pull request merged**